content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,612 @@
+---
+slug: 35 title: 35/WAKU2-BINDINGS-API
+name: Waku v2 C Bindings API
+status: draft
+tags: waku-core
+editor: Richard Ramos <richard@status.im>
+contributors:
+- Franck Royer <franck@status.im>
+---
+
+# Introduction
+
+Native applications that wish to integrate Waku may not be able to use nim-waku and its JSON RPC API due to constraints
+on packaging, performance or executables.
+
+An alternative is to link existing Waku implementation as a static or dynamic library in their application.
+
+This specification describes the C API that SHOULD be implemented by native Waku library and that SHOULD be used to
+consume them.
+
+# Design requirements
+
+The API should be generic enough, so:
+
+- it can be implemented by both nim-waku and go-waku C-Bindings,
+- it can be consumed from a variety of languages such as C#, Kotlin, Swift, Rust, C++, etc.
+
+The selected format to pass data to and from the API is `JSON`.
+
+It has been selected due to its widespread usage and easiness of use. Other alternatives MAY replace it in the future (C
+structure, protobuf) if it brings limitations that need to be lifted.
+
+# The API
+
+## General
+
+### JsonResponse
+
+All the API functions return a `JsonResponse` unless specified otherwise.
+`JsonResponse` is a `char *` whose format depends on whether the function was executed successfully or not.
+
+On failure:
+
+```json
+{
+  "error": "the error message"
+}
+```
+
+On success:
+
+```json
+{
+  "result": {}
+}
+```
+
+The format of the `result` object depends on the function it was returned by. 
+
+## Events
+
+Asynchronous events require a callback to be registered.
+An example of an asynchronous event that might be emitted is receiving a message.
+When an event is emitted, this callback will be triggered receiving a json string with the following format:
+
+```json
+{
+  "nodeId": 0, // go-waku node that emitted the signal
+  "type": "message", // type of signal being emitted. Currently only "message" is available
+  "event": ... // format depends on the type of signal. In the case of "message", a waku message can be expected.
+}
+```
+
+| `type` | `event` Type  |
+|:------|---------------|
+| `message` | TODO      | 
+
+
+### `extern void waku_set_event_callback(void* cb)`
+
+Register callback to act as event handler and receive application signals,
+which are used to react to asynchronous events in Waku.
+
+**Parameters**
+
+1. `void* cb`: callback that will be executed when an async event is emitted.
+  The function signature for the callback should be `void myCallback(char* signalJSON)`
```

Comment:
Is `signalJson` the type described in the `events` section? @richard-ramos

Reply:
Yes